File: src/main/java/um/haberes/rest/model/CargoTipo.java
/**
 * 
 */
package um.haberes.rest.model;

import java.io.Serializable;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;

import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

/**
 * @author daniel
 *
 */
@Entity
@EqualsAndHashCode(callSuper = false)
@Getter
@Setter
@ToString
@NoArgsConstructor
@AllArgsConstructor
public class CargoTipo extends Auditable implements Serializable {
	/**
	 * 
	 */
	private static final long serialVersionUID = 3305190933765537796L;

	@Id
	private Integer cargoTipoId;

	private Byte aCargo = 0;
	private String nombre = "";
	private Integer precedencia = 0;

}

File: src/main/java/um/haberes/rest/kotlin/model/CargoTipo.kt
package um.haberes.rest.kotlin.model

import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table
import um.haberes.rest.model.Auditable

@Entity
@Table
data class CargoTipo(

    @Id
    var cargoTipoId: Int? = null,

    var aCargo: Byte = 0,
    var nombre: String = "",
    var precedencia: Int = 0

) : Auditable()
